src/m_misc.c

```diff
@@ -83,6 +83,72 @@ boolean M_FileExists(char *filename)
     }
 }
 
+char *M_FileCaseExists(char *path)
```

Comment:
Can you add a short comment before this function documenting its behavior and specifically the fact that it returns a newly allocated string that the caller is responsible for freeing?